src/core/reactor.cc

```diff
@@ -2200,6 +2202,35 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<std::optional<struct group>> reactor::getgrnam(std::string_view name, char *buf, size_t buflen) {
+    syscall_result_extra<std::optional<struct group>> sr = co_await _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>(
+        [name = sstring(name), buf, buflen] {
```

Comment:
@nyh , I went with the copy approach, since most of the other function calls were also copying to new sstring.

And when I avoid copying and just use `name`, I see this compile error:

```
[1/5] Building CXX object CMakeFiles/seastar.dir/src/core/reactor.cc.o
FAILED: CMakeFiles/seastar.dir/src/core/reactor.cc.o 
/usr/local/bin/clang++ -DFMT_SHARED -DSEASTAR_API_LEVEL=7 -DSEASTAR_BUILD_SHARED_LIBS -DSEASTAR_DEFERRED_ACTION_REQUIRE_NOEXCEPT -DSEASTAR_DEPRECATED_OSTREAM_FORMATTERS -DSEASTAR_ENABLE_ALLOC_FAILURE_INJECTION -DSEASTAR_HAS_MEMBARRIER -DSEASTAR_HAVE_ASAN_FIBER_SUPPORT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_SYSTEMTAP_SDT -DSEASTAR_HAVE_URING -DSEASTAR_LOGGER_COMPILE_TIME_FMT -DSEASTAR_LOGGER_TYPE_STDOUT -DSEASTAR_PTHREAD_ATTR_SETAFFINITY_NP -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_SSTRING -DSEASTAR_STRERROR_R_CHAR_P -DSEASTAR_THREAD_STACK_GUARDS -DSEASTAR_TYPE_ERASE_MORE -Dseastar_EXPORTS -I/home/raj/db/scylla/seastar/include -I/home/raj/db/scylla/seastar/build/dev/gen/include -I/home/raj/db/scylla/seastar/build/dev/gen/src -I/home/raj/db/scylla/seastar/src -O2 -std=gnu++23 -fPIC -U_FORTIFY_SOURCE -Wno-error=unused-result "-Wno-error=#warnings" -fstack-clash-protection -UNDEBUG -Wall -Werror -Wimplicit-fallthrough -Wdeprecated -Wno-error=deprecated -ftls-model=initial-exec -gz -MD -MT CMakeFiles/seastar.dir/src/core/reactor.cc.o -MF CMakeFiles/seastar.dir/src/core/reactor.cc.o.d -o CMakeFiles/seastar.dir/src/core/reactor.cc.o -c /home/raj/db/scylla/seastar/src/core/reactor.cc
/home/raj/db/scylla/seastar/src/core/reactor.cc:2212:41: error: no member named 'c_str' in 'std::basic_string_view<char>'
 2212 |             int ret = ::getgrnam_r(name.c_str(), &grp, buf, buflen, &result);
      |                                    ~~~~ ^
In file included from /home/raj/db/scylla/seastar/src/core/reactor.cc:181:
/home/raj/db/scylla/seastar/src/core/thread_pool.hh:44:32: error: no matching member function for call to 'submit'
   44 |         return inter_thread_wq.submit<T>(std::move(func));
      |                ~~~~~~~~~~~~~~~~^~~~~~~~~
/home/raj/db/scylla/seastar/src/core/reactor.cc:2206:83: note: in instantiation of function template specialization 'seastar::thread_pool::submit<seastar::syscall_result_extra<std::optional<group>>, (lambda at /home/raj/db/scylla/seastar/src/core/reactor.cc:2207:9)>' requested here
 2206 |     syscall_result_extra<std::optional<struct group>> sr = co_await _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>(
      |                                                                                   ^
/home/raj/db/scylla/seastar/src/core/syscall_work_queue.hh:65:15: note: candidate function template not viable: no known conversion from 'typename std::remove_reference<(lambda at /home/raj/db/scylla/seastar/src/core/reactor.cc:2207:9) &>::type' (aka '(lambda at /home/raj/db/scylla/seastar/src/core/reactor.cc:2207:9)') to 'noncopyable_function<syscall_result_extra<optional<group>> ()>' for 1st argument
   65 |     future<T> submit(noncopyable_function<T ()> func) noexcept {
      |               ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
ninja: build stopped: subcommand failed.

```

Reply:
Ok, now I understand why these copies were done. std::string_view does not have a c_str() method because it is not guaranteed to be null-terminated. To get a null-terminated string, which we need for the C library function, the easiest thing is to copy to a std::string or seastar::sstring and use its c_str().
So your code is fine and I'm resolving this issue.